components/table/Column.razor.cs

```diff
@@ -191,8 +191,11 @@ private void SetSorter(SortDirection sortDirection)
 
         private void ToggleTreeNode()
         {
+            bool expandValueBeforeChange = RowData.Expanded;
             RowData.Expanded = !RowData.Expanded;
-            Table?.Refresh();
+            Table?.OnExpandChange(RowData.CacheKey);
+            if (RowData.Expanded != expandValueBeforeChange)
+                Table?.Refresh();
```

Comment:
RowData.Expanded is always not equal to expandValueBeforeChange.

Reply:
That is the idea. My use case is that I am checking some kind of conditions and do not allow expansion if they are not met. So if the expand was clicked but consumer's code changed it, there is no need to refresh anything.

Reply:
To some degree, if you wanted, you could probably simulate accordion functionality with this.

Reply:
Ok, get it.